sky/serve/load_balancer.py

```diff
@@ -70,30 +76,115 @@ def _sync_with_controller(self):
+        try:
+            # We defer the get of the client here on purpose, for case when the
+            # replica is ready in `_proxy_with_retries` but refreshed before
+            # entering this function. In that case we will return an error here
+            # and retry to find next ready replica. We also need to wait for the
+            # update of the client pool to finish before getting the client.
+            with self._client_pool_lock:
+                client = self._client_pool.get(url, None)
+            if client is None:
+                return RuntimeError(f'Client for {url} not found.')
+            worker_url = httpx.URL(path=request.url.path,
+                                   query=request.url.query.encode('utf-8'))
+            proxy_request = client.build_request(
+                request.method,
+                worker_url,
+                headers=request.headers.raw,
+                content=await request.body(),
+                timeout=constants.LB_STREAM_TIMEOUT)
+            proxy_response = await client.send(proxy_request, stream=True)
+            return fastapi.responses.StreamingResponse(
+                content=proxy_response.aiter_raw(),
+                status_code=proxy_response.status_code,
+                headers=proxy_response.headers,
+                background=background.BackgroundTask(proxy_response.aclose))
+        except (httpx.RequestError, httpx.HTTPStatusError) as e:
+            logger.error(f'Error when proxy request to {url}: '
+                         f'{common_utils.format_exception(e)}')
+            return e
+
+    async def _proxy_with_retries(
+            self, request: fastapi.Request) -> fastapi.responses.Response:
+        """Try to proxy the request to the endpoint replica with retries."""
+        self._request_aggregator.add(request)
+        # TODO(tian): Finetune backoff parameters.
+        backoff = common_utils.Backoff(initial_backoff=1)
+        # SkyServe supports serving on Spot Instances. To avoid preemptions
+        # during request handling, we add a retry here.
+        retry_cnt = 0
+        while True:
+            retry_cnt += 1
+            with self._client_pool_lock:
+                ready_replica_url = self._load_balancing_policy.select_replica(
+                    request)
+            if ready_replica_url is None:
+                raise fastapi.HTTPException(
+                    # 503 means that the server is currently
+                    # unable to handle the incoming requests.
+                    status_code=503,
+                    detail='No ready replicas. '
+                    'Use "sky serve status [SERVICE_NAME]" '
+                    'to check the replica status.')
+            response_or_exception = await self._proxy_request_to(
+                ready_replica_url, request)
+            if not isinstance(response_or_exception, Exception):
+                return response_or_exception
+            if await request.is_disconnected():
```

Comment:
Add a comment for abortion, please double check the correctness.
```suggestion
            # When the user aborts the request during streaming, the request will be disconnected.
            # We do not need to retry for this case.
            if await request.is_disconnected():
```